### ISSUE TITLE
ffmpeg running after all streams are closed

### ISSUE THREAD

**Issue by jcastilloalonso** [tgorgdotcom/locast2plex#17]
Hey, been running locast2plex for a couple of days. It's been running great so far. Today doing some updates to my Plex server, I usually like to check on process list and make sure nothing is going crazy. Upon doing ps aux I was surprised to see some ffmpeg process running:

root     26417  0.0  0.1 369520 19164 pts/0    S+   Aug07   0:02 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/WmnVO7nRaDrM7zYfWQpubbLNSZ8siPmxKst3lbErmELXbr8A9XyHuvfbmNYN0ZAi4r5aLd
kZZOr7LdWVdh7EXy.m3u8 -codec copy -f mpegts pipe:1
root     26638  0.0  0.1 369152 17948 pts/0    S+   Aug07   0:18 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/PtrlkUL6b0gl1y9fXl6dfycqVds6CTp8C73Zf6Fp5IWbmOoJ9DJtiiOvByoCfyPzf4yop3
zbTQpBbHIZfOCLGi.m3u8 -codec copy -f mpegts pipe:1
root     26325  0.0  0.1 369112 16652 pts/0    S+   Aug07   0:00 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/rdsIP1zris6zTKjJtK6Pn1MKt2m33JJiMQp3cTECLfAGAMWbSFsHn4aZv35ybhug6UpDaf
wvAsQw8COF3FZuro.m3u8 -codec copy -f mpegts pipe:1
root     27850  0.0  0.1 369492 16120 pts/0    S+   Aug07   0:00 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/L1ADmJ791i7xjeYXzT2tiJKKe2rpcAIyqz756usf7vJtKZjgL80KfeqV2aRaTfM1DFTV1t
kEIqnW4gjErqyo5C.m3u8 -codec copy -f mpegts pipe:1
root      6536  0.0  0.1 369316 16080 pts/0    S+   Aug08   0:37 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/AwFG5BaaEpZlQaTdccQNGtOEOcgSLQ57dtCwrXRTq7CAdyEWp39H2bD3mpc6cXE8zIhroA
02M1B3rwaQpKBxFq.m3u8 -codec copy -f mpegts pipe:1
root      3707  0.0  0.1 369524 16064 pts/0    S+   Aug07   0:03 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/jqynS11ka18Yk5wDzwk6cKpDcCAY0AX1j7tHp2d0qa2acLH8yFSa8F8Nt9pUHJAdsSb9UD
iJXLmKvfh9jFOQCB.m3u8 -codec copy -f mpegts pipe:1
root     32589  0.0  0.1 369544 16000 pts/0    S+   Aug07   0:04 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/Wh14Bxo1japZ26EgqWsUTedRrkEdkPMjS8TKD52LtjB8uIzMLNIWqWgLi1LDENHZlOcI1R
hkbngYAZETnG1rc0.m3u8 -codec copy -f mpegts pipe:1
root     32564  0.0  0.1 369472 15736 pts/0    S+   Aug07   0:00 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/DEqxz2Yzh0u8cruJ8kCx7lIJNPfbNLsnbhLuzknegKnWt4BmMs01HzRGcijjIYtp5nLqfX
ieSf3EFWfVEEGJKS.m3u8 -codec copy -f mpegts pipe:1
root     26338  0.0  0.1 369464 15600 pts/0    S+   Aug07   0:00 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/TxtkOtvhXvLQ44ghoAVpNVZn42ai7kAYM8ieNgm1K3AuyxRblhjMTDcgTKF58RLwwdqrIt
oZc9GB3oMvwOMZiF.m3u8 -codec copy -f mpegts pipe:1
root      3657  0.0  0.1 369380 15508 pts/0    S+   Aug08   0:00 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/5lnyIO8iCUPd9sDGtiF2D2oI8RiH73JJko3jzPRiBW2IXY9DQZE5ENCVryxAjVWs0xj1WK
mL1J0K2IPQdotX8anOkcjSYTuAJbRFplRjyHFW.m3u8 -codec copy -f mpegts pipe:1
root     29144  0.0  0.1 368472 15300 pts/0    S+   Aug07   0:02 ffmpeg -i https://dal.locastnet.org/proxy/dal/variant/3zjwqgfwLQSK08UYIIdQMcXSNT8JOLiQE0PK5AOvarmmd2sPMHBNcyAMSQy4eYudvoWm9m
QYzYq0Si7hP0KwM3IfGrQik0AzERv4RTbxQS5u.m3u8 -codec copy -f mpegts pipe:1

I was not watching anything over at Plex, and as you can see some of those are over 2 days old. I glimpsed over the debug log and couldn't find anything that popped my eye. 

Also, for future troubleshooting, do you allow saving the log to a file? This will make debugging some issues way easier than printing over to the screen.

Thanks a lot for your tool! It's really amazing to be able to DVR the content and use locast as an alternative as currently an antenna in my home is impossible.

Keep up the amazing work!

**Comment by jcastilloalonso** [comment]
Not sure if related but noticed this exception in my logs today:

127.0.0.1 - - [10/Aug/2020 12:13:37] "GET /lineup_status.json HTTP/1.0" 200 -
[hls,applehttp @ 0x241ee80] Opening 'https://dal.locastnet.org/proxy/dal/ktvt_src266179946.ts' for reading
----------------------------------------3kB time=01:14:17.98 bitrate=1510.2kbits/s speed=   1x
Exception happened during processing of request from ('127.0.0.1', 58328)
Traceback (most recent call last):
  File "/usr/lib64/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/usr/lib64/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/usr/lib64/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/usr/lib64/python2.7/SocketServer.py", line 651, in __init__
    self.finish()
  File "/usr/lib64/python2.7/SocketServer.py", line 710, in finish
    self.wfile.close()
  File "/usr/lib64/python2.7/socket.py", line 279, in close
    self.flush()
  File "/usr/lib64/python2.7/socket.py", line 303, in flush
    self._sock.sendall(view[write_offset:write_offset+buffer_size])
error: [Errno 32] Broken pipe
----------------------------------------

**Comment by jcastilloalonso** [comment]
I've been able to recreate this over and over. Using CentOS 7. It seems there is a bug closing the stream. This ends even breaking bash, forcing me to close the screen session to fix it. Please let me know if I can provide any additional information

**Comment by scumola** [comment]
I'm getting this too.  Running Ubuntu 16.04 w/ the python (non-docker) version.  Streams close when I stop the python script, but many open ffmpeg streams after a few days or recording.
```
steve    29126  0.0  0.2 442204 21304 pts/4    D    Aug15   0:00 ffmpeg -i https://den.locastnet.org/proxy/den/variant/6AKP5kfkmwqXUMN9NtirEtFGgCYj3LCjgE4XMOqdGtqhUhTsfwhZxoMUVftvj529xlYFexRwPTIn4DPabsNjTOH1W6G6d3d4khoqSgeOd1Zw.m3u8 -codec copy -f mpegts pipe:1
steve    30980  0.0  0.1 441876 12348 pts/4    D    Aug13   0:00 ffmpeg -i https://den.locastnet.org/proxy/den/variant/lkdRZf8uG9h15sIZ6jPStucBBpQ9JN3i886upbOeHj7WiNPxUwYoU6dgbdAUfAtifOtuqjLbWJj5GNL24kdvTk.m3u8 -codec copy -f mpegts pipe:1
steve    31142  0.0  0.1 442104 13684 pts/4    D    Aug14   0:03 ffmpeg -i https://den.locastnet.org/proxy/den/variant/3qtEP09PUYbG8pw3xJ9DziiZuOVCkNwSNoOiyFwRNlJkf8LSaznPQoCNZCGGefvLzWXTykl7aOl0lWejj2XHbRzFPZXGjxeHrPyvqXgrrZgV.m3u8 -codec copy -f mpegts pipe:1
```

**Comment by ratherDashing** [comment]
Same here with the docker image.  I have to stop/start the container every few days or else LiveTV becomes unstable.

**Comment by jcastilloalonso** [comment]
Closing this issue as it's been solved with PR #44
